.ado/templates/apple-xcode-build.yml

```diff
@@ -21,7 +21,7 @@ steps:
       signingOption: auto
       packageApp: false
       teamId: '$(XCodeSigningMicrosoftTeamID)'
-      args: '-destination "${{ parameters.xcode_destination }}" ONLY_ACTIVE_ARCH=NO -verbose -UseModernBuildSystem=NO -derivedDataPath DerivedData ${{ parameters.xcode_extraArgs }}'
```

Comment:
@HeyImChris was there any particular reason you know of we were forcing legacy build system? I probably should have done things one at a time to control variables but this was a delta between CI and building locally in Xcode by default so seemed helpful to get rid of that to make it easier to reproduce CI issues?

Reply:
The legacy build system used to be required before we built using cocoapods.   The old static xcodeproj's from upstream had intentional circular dependencies that were worked around by having build order in the xcschemes.   The new pod based builds don't have this issue anymore.